pkg/rpc/server/subscription_test.go

```diff
@@ -109,23 +118,104 @@ func TestSubscriptions(t *testing.T) {
 	}
 
 	for _, id := range subIDs {
-		var b bool
-
-		resp := callWSGetRaw(t, c, fmt.Sprintf(`{
-  "jsonrpc": "2.0",
-  "method": "unsubscribe",
-  "params": ["%s"],
-  "id": 1
-}`, id), respMsgs)
-		require.Nil(t, resp.Error)
-		require.NotNil(t, resp.Result)
-		require.NoError(t, json.Unmarshal(resp.Result, &b))
-		require.Equal(t, true, b)
+		callUnsubscribe(t, c, respMsgs, id)
 	}
 	finishedFlag.CAS(false, true)
 	c.Close()
 }
 
+func TestFilteredSubscriptions(t *testing.T) {
+	var cases = map[string]struct {
+		params string
+		check  func(*testing.T, *response.Notification)
+	}{
+		"tx matching": {
+			params: `["transaction_added", {"type":"InvocationTransaction"}]`,
+			check: func(t *testing.T, resp *response.Notification) {
+				rmap := resp.Payload[0].(map[string]interface{})
```

Comment:
This is a server test and it doesn't try to interpret these JSONs as real structures (it would complicate this test probably), instead it uses the same type server uses for marshaling and that has plain `interface{}` that is unmarshalled in standard `encoding/json` fashion.